flax/core/meta.py

```diff
@@ -287,6 +288,21 @@ def get_sharding(self, mesh: jax.sharding.Mesh) -> jax.sharding.Sharding:
     """Returns the ``NamedSharding`` for this partitioned value."""
     return jax.sharding.NamedSharding(mesh, self.get_partition_spec())
 
+  def to_nnx_metadata(self) -> dict[str, Any]:
+    """Return a dict of metadata that can translate into an `nnx.Variable`."""
+    metadata = vars(self)
+    metadata['sharding'] = metadata['names']
+    del metadata['names']
+    return metadata
+
+  @classmethod
+  def from_nnx_metadata(cls, metadata: dict[str, Any]):
+    """Given a dict of `nnx.Variable` format metadata, create a `nn.Partitioned`."""
+    metadata['names'] = metadata['sharding']
+    del metadata['sharding']
+    fields = [x.name for x in dataclasses.fields(cls)]
```

Comment:
feels like `fields` should be `set`

Reply:
I think `dataclasses.fields` should already return de-duplicated names, otherwise the class cannot be built... but yeah we can add it

Reply:
oh I mean because in the line below we just use it to check for membership